knowledge_engine/tests/test_retrieval_filters.py

```
@@ -78,8 +100,125 @@ def test_parse_metadata_filters_normalizes_operator_aliases() -> None:
+                "conditions": [
+                    {"key": "status", "operator": "==", "value": "archived"}
+                ],
+            },
+        )
+
+
+@pytest.mark.parametrize("operator", ["eq", "in", "contains"])
+def test_parse_metadata_filters_rejects_document_scope_metadata(
+    operator: str,
+) -> None:
+    with pytest.raises(ValueError, match="RetrievalScope.document_ids"):
```

Ruff (0.15.20)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)